BusinessLayer/Services/DepartmentOptionService.cs

```diff
@@ -0,0 +1,134 @@
+﻿using BusinessLayer.Interface;
+using DataLayer.Dtos;
+using DataLayer.Model;
+using Microsoft.Extensions.Configuration;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+using System.Linq;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.AspNetCore.Http;
+
+namespace BusinessLayer.Services
+{
+    public class DepartmentOptionService : Repository<DepartmentOption>, IDepartmentOptionService
+    {
+        private readonly IConfiguration _configuration;
+        public readonly string baseUrl;
+        ResponseModel response= new ResponseModel();
+
+        public DepartmentOptionService(FPNOOCRContext context, IConfiguration configuration) : base(context)
+        {
+            _configuration = configuration;
+            baseUrl = configuration.GetValue<string>("Url:root");
+        }
+
+        public async Task<ResponseModel> AddDepartmentOption(DepartmentOptionDto model)
+        {
+            try
+            {
+                DepartmentOption deptOption = new DepartmentOption();
+                //var d_slug = Utility.GenerateSlug(model.Name);
+                var doesExist = await _context.DEPARTMENT_OPTION.Where(dpo => dpo.Id == model.Id).FirstOrDefaultAsync();
+                if(doesExist != null)
+                {
+                    response.StatusCode = StatusCodes.Status208AlreadyReported;
+                    response.Message = "Department Option Already Added";
+                    return response;
+                }
+                deptOption.Name = model.Name;
+                deptOption.Active = true;
+                deptOption.DepartmentId = model.DepartmentId;
+               // dept.DateCreated = DateTime.Now;
+                _context.Add(deptOption);
+                await _context.SaveChangesAsync();
+                return response;
+            }
+            catch (Exception ex)
+            {
+
+                throw;
+            }
+        }
+
+        public async Task<ResponseModel> UpdateDepartmentOption(DepartmentOptionDto model)
+        {
+            try
+            {
+                DepartmentOption deptOption = await _context.DEPARTMENT_OPTION.Where(f => f.Id == model.Id).FirstOrDefaultAsync();
+                if (deptOption == null)
+                {
+                    throw new NullReferenceException("Department Option not found");
```

Comment:
This is wrong.. What happens if departmentOption is not equal to null? I strongly suggest you extract line 63 - 70 and put them in an else block because from what you wrote here, this method does nothing when departmentOption != null